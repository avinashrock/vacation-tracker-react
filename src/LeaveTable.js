import React,{Component} from 'react';
import {Table} from 'react-bootstrap';
import swal from 'sweetalert';
class LeaveTable extends Component{
    constructor(props){
        super(props);
        this.state = {
            key:0
        }
        this.onAccept = this.onAccept.bind(this);
        this.onReject = this.onReject.bind(this);
    }
    componentWillReceiveProps(){
        this.render();
    }
    onAccept = (e) =>{
        this.setState({key:e.target.value});
        swal({
            title: "Leave got accepted",
            icon: "success",
            confirmButtonText: 'OK',
        }).then((value)=>{
            let empid = document.getElementsByClassName('accept');
            value = localStorage.getItem(empid[this.state.key].value);
            swal(`your leaveId is ${value}`);
            let store = document.getElementsByClassName('detail');
            store[this.state.key].style.display = 'none';
        });
    }
    onReject = (e)=>{
        this.setState({key:e.target.value});
        swal({
            title: "Sorry, Your leave was not accepted",
            icon: "warning",
            confirmButtonText: 'OK',
        }).then((value)=>{
            let empid = document.getElementsByClassName('reject');
            value = localStorage.getItem(empid[this.state.key].value);
            swal(`your leaveId is ${value}`);
            let store = document.getElementsByClassName('detail');
            store[this.state.key].style.display = 'none';
        });
    }
    render(){
        if(this.props.data !== null && this.props.data.length > 0) {
            const rows = this.props.data.map((val, i) => {
                localStorage.setItem(i,val.leaveId)
                if(val.leaveStatus===0){
                    val.leaveStatus='Pending'
                }
                return <tr key={i} className="detail" >
                            <td>{val.leaveId}</td>
                            <td>{val.name}</td>
                            <td>{val.startDate}</td>
                            <td>{val.endDate}</td>
                            <td>{val.reason}</td>
                            <td>{val.leaveStatus}</td>
                            <td><button id='accept' className="accept" value={i} onClick={this.onAccept}>Accept</button></td>
                            <td><button id='reject' className="reject" value={i} onClick={this.onReject}>Reject</button></td>
    
                       </tr>
            })
            return (
                <Table responsive striped bordered condensed hover>
                    <thead>
                        <tr>
                            <th>LeaveId</th>
                            <th>Name</th>
                            <th>StartDate</th>
                            <th>EndDate</th>
                            <th>Reason</th>
                            <th>LeaveStatus</th>
                            <th>Accept</th>
                            <th>Reject</th>
                        </tr>
                    </thead>
                    <tbody>
                        {rows}
                    </tbody>
                </Table>
            );
    }
    else {
        return (
            <section></section>
        );
    }

}
}
export default LeaveTable;