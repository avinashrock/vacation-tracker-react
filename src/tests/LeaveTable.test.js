import React from 'react';
import {shallow,mount} from 'enzyme';
import LeaveTable from '../LeaveTable';


const props = {
    data:{
        leaveId:0,
        userId:'AP062035',
        startDate:'2018-09-10',
        endDate:'2018-09-11',
        reason:'asassd',
        status:0,
        }
}
const wrapper = shallow(<LeaveTable {...props}/>);
describe('component renders without crashing',()=>{
    it('should render properly',()=>{
        expect(wrapper.exists()).toBe(true);
    })
    it('should match snapshot',()=>{
        expect(wrapper).toMatchSnapshot();
    })
    it('should handle state changes ', () => {
        expect(wrapper.state().key).toEqual(0);
        wrapper.simulate('click',{target:{key:1}});
    })
    it('lifecycle method exists',()=>{
        wrapper.instance().componentWillReceiveProps();
    })
    // it('onAccept',()=>{
    //     wrapper.instance().onAccept();
    // })
    
})