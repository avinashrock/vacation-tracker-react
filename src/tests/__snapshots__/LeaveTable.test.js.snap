// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders without crashing should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LeaveTable
    data={
        Object {
            "endDate": "2018-09-11",
            "leaveId": 0,
            "reason": "asassd",
            "startDate": "2018-09-10",
            "status": 0,
            "userId": "AP062035",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
